service/history/shard/context_impl.go

```diff
@@ -1949,6 +1985,10 @@ func (s *ContextImpl) acquireShard() {
 		// we know engineFuture must be ready here, and we can notify queue processor
 		// to trigger a load as queue max level can be updated to a newer value
 		s.notifyQueueProcessor()
+		// This runs until the lifecycleCtx is cancelled, so we only need to start it once
+		if s.emittingQueueMetrics.CompareAndSwap(false, true) {
+			go s.monitorQueueMetrics()
+		}
 
 		s.updateHandoverNamespacePendingTaskID()
```

Comment:
nit: I feel like we need to create a separate func for all those steps (including those re. to engineFuture that are already in s.transition) and call it within s.transition when the new state is Acquired.

Reply:
I'll do that in a follow-up PR to keep this one focused, if that's alright